### PR TITLE
Update developer-setup.md to provide no-docker setup

### PR DESCRIPTION
**Pull request by sinansonmez**
#### Summary
Update developer-setup.md to provide no-docker setup information

### CODE REVIEW

File: site/content/contribute/developer-setup.md [hunk 1]
```diff
@@ -36,7 +36,16 @@ If you're using Windows, we recommend using the Windows Subsystem for Linux (WSL
    sudo apt install build-essential
    ```
 
-1. Install [Docker](https://www.docker.com/).
+1. Install [Docker](https://www.docker.com/). If you dont want to use Docker, you can follow the steps in the sub-bullets for manual database setup
```
Comment (cwarnermm) [inline site/content/contribute/developer-setup.md:39]:
```suggestion
1. Install [Docker](https://www.docker.com/). If you don't want to use Docker, you can follow the steps in the sub-bullets below for manual database setup.
```

File: site/content/contribute/developer-setup.md [hunk 2]
```diff
@@ -36,7 +36,16 @@ If you're using Windows, we recommend using the Windows Subsystem for Linux (WSL
...
+    1. Zet `MM_NO_DOCKER` to `true` in `server/config.mk` file
```
Comment (cwarnermm) [inline site/content/contribute/developer-setup.md:40]:
```suggestion
    1. Set `MM_NO_DOCKER` to `true` in the `server/config.mk` file.
```

File: site/content/contribute/developer-setup.md [hunk 3]
```diff
@@ -36,7 +36,16 @@ If you're using Windows, we recommend using the Windows Subsystem for Linux (WSL
...
+    2. Install [PostgreSQL](https://www.postgresql.org/download/)
+    3. Run `psql postgres`. Then create `mmuser` by running `CREATE ROLE mmuser WITH LOGIN PASSWORD 'mostest';`
+    4. Modify the role to give rights to create ` database by running `ALTER ROLE mmuser CREATEDB;`
```
Comment (cwarnermm) [inline site/content/contribute/developer-setup.md:43]:
```suggestion
    4. Modify the role to give rights to create a database by running `ALTER ROLE mmuser CREATEDB;`
```

File: site/content/contribute/developer-setup.md [hunk 4]
```diff
@@ -36,7 +36,16 @@ If you're using Windows, we recommend using the Windows Subsystem for Linux (WSL
...
+    5. Check the role by running `\du`
```
Comment (cwarnermm) [inline site/content/contribute/developer-setup.md:44]:
```suggestion
    5. Confirm the role rights by running `\du`
```

File: site/content/contribute/developer-setup.md [hunk 5]
```diff
@@ -36,7 +36,16 @@ If you're using Windows, we recommend using the Windows Subsystem for Linux (WSL
...
+    6. Before creating the database, exit by running `\q`
+    7. Login again via mmuser by running `psql postgres -U mmuser`
```
Comment (cwarnermm) [inline site/content/contribute/developer-setup.md:46]:
```suggestion
    7. Login again via `mmuser` by running `psql postgres -U mmuser`
```

**Review by cwarnermm — CHANGES_REQUESTED** [review]
Thanks, @sinansonmez. Provided editorial feedback.

**Review by hmhealey — CHANGES_REQUESTED** [review]
When you make those changes, could you move those steps to a separate section at the bottom of the doc and change that "Install Docker" step to point the user to the new section if they don't want to use Docker? I think we should split those steps into their own section because I think people will miss that those steps are optional and go through them after installing Docker. I want to keep the main instructions simpler because I screw up complicated instructions all the time 😅

**Comment by sinansonmez** [comment]
> When you make those changes, could you move those steps to a separate section at the bottom of the doc and change that "Install Docker" step to point the user to the new section if they don't want to use Docker? I think we should split those steps into their own section because I think people will miss that those steps are optional and go through them after installing Docker. I want to keep the main instructions simpler because I screw up complicated instructions all the time 😅

@hmhealey @cwarnermm Thanks for the feedback. Unfortunately, it is not clear to me how to proceed. 

Currently there are 3 main sections
- Set up your development environment 
- Prerequisites for Windows
- Setup the Mattermost Server 

Under `Setup the Mattermost Server ` there are 13 steps. Should I create a new main section or add the 14th step under `Setup the Mattermost Server ` ?

**Comment by cwarnermm** [comment]
Hi @sinansonmez! The ask is to separate the Docker steps within their own section (at the top of the page as the primary way forward) from the not-using-Docker steps you're providing. @hmhealey is suggesting we move the non-Docker info to its own section elsewhere on the page to avoid reader confusion.

The reader's experience would look like this:
- Reader reaches the **Setup the Mattermost Server** section.
- Step 2 instructs them to install Docker. Step 2 also includes a link to the new non-Docker section later on the page in case the reader would prefer that approach.
- The rest of this section is about using Docker.

Meanwhile, following the **Setup the Mattermost Server** section, there's a new section about setting up Mattermost Server without Docker. That's where your steps would live.

**Comment by hmhealey** [comment]
Yep, that's right. So with your steps, there'd be 4 sections:
- Set up your development environment
- Prerequisites for Windows
- Setup the Mattermost Server
- Develop Mattermost without Docker

**Comment by sinansonmez** [comment]
thanks for the explanation @hmhealey @cwarnermm the page is updated as requested

**Review by cwarnermm — APPROVED** [review]
Thanks, @sinansonmez!

**Review by hmhealey — APPROVED** [review]
Thanks, Sinan!

**Pull request merged**